src/config/ActiveSettings.cpp

```diff
@@ -220,8 +220,63 @@ fs::path ActiveSettings::GetMlcPath()
 	return GetDefaultMLCPath();
 }
 
+fs::path ActiveSettings::GetConfigPath() {
+#ifdef XDG
+    auto config_home = std::getenv("XDG_CONFIG_HOME");
+    auto config_home_s = std::string();
+    if (!config_home) {
+        config_home_s = fmt::format("{}/.config", std::getenv("HOME"));
+    }
+    auto dir = fs::path(fmt::format("{}/cemu", config_home ? config_home : config_home_s));
+    if (!fs::exists(dir)) {
+      fs::create_directories(dir);
+    }
+    return dir;
```

Comment:
`std::getenv("XDG_CONFIG_HOME")` will evaluate to false if `XDG_CONFIG_HOME` is unset (as `getenv` returns `nullptr` in this case), but what if it is set to an empty string?

Also, you are checking if `config_home` is set twice. What about:

```suggestion
	auto config_home = std::getenv("XDG_CONFIG_HOME");
	auto dir = fs::path(fmt::format(
		"{}/cemu",
		config_home ? config_home : fmt::format("{}/.config", std::getenv("HOME"))
	));
	if (!fs::exists(dir)) {
		fs::create_directories(dir);
	}
	return dir;
```

Reply:
Should probably make a wrapper function that handles these cases and always returns `std::string`.

Reply:
I guess that a simple function like this should be enough

```cpp
std::string get_xdg_dir(const std::string_view name) {
	std::string config_home = std::getenv(name);
	if (config_home.empty()) {
		config_home = fmt::format("{}/.config", std::getenv("HOME"));
	}
	return config_home;
}
```